.github/workflows/deploy_ecr.yml

```diff
@@ -0,0 +1,40 @@
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ dev ]
```

Comment:
the idea is that you build and push the image on successful merge to your main branch. this is achieved by `push`. why you need `pull_request`? with this condition it's gonna be treggered on pull request open and close https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request